fe/fe-core/src/main/java/com/starrocks/sql/optimizer/CachingMvPlanContextBuilder.java

```diff
@@ -33,6 +44,38 @@ public class CachingMvPlanContextBuilder {
 
     private Cache<MaterializedView, List<MvPlanContext>> mvPlanContextCache = buildCache();
 
+    public static class AstKey {
+        private final ParseNode parseNode;
+        public AstKey(ParseNode parseNode) {
+            this.parseNode = parseNode;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || ! (o instanceof AstKey)) {
+                return false;
+            }
+            AstKey other = (AstKey) o;
+            // TODO: add more checks.
+            return this.toString().equals(other.toString());
```

Comment:
compare the AST would be much more efficient if two AST are different:
- A: `select .... where id = 1`, B: `select ..... where id = 2`